File: kotlin/Class1.kt
package com.example

open class BaseClass {
    protected val baseVar = "I'm a base class variable"

    open suspend fun baseMethod() {
        println("This is a method from BaseClass")
        println("Base var: $baseVar")
        kotlinx.coroutines.delay(100)
    }
}

class DerivedClass : BaseClass() {
    private val derivedVar = "I'm a derived class variable"

    suspend fun derivedMethod() {
        println("This is a method from DerivedClass")
        println("Derived var: $derivedVar")
        baseMethod()
    }

    override suspend fun baseMethod() {
        super.baseMethod()
        println("This is an overridden method in DerivedClass")
    }
}

class Class1 {
    private val privateVar = "I'm private"

    suspend fun method1() {
        println("This is method1 from Class1")
        privateMethod()
    }

    private suspend fun privateMethod() {
        println("This is a private method in Class1")
        println("Private var: $privateVar")
        kotlinx.coroutines.delay(100) // Simulate some async work
    }

    companion object {
        const val STATIC_VAR = "I'm a static variable in Class1"
    }
}

File: csharp/Class1.cs
using System;
using System.Threading.Tasks;
using System.Collections.Generic;

namespace CSharpExample
{
    public class Class1
    {
        private string privateVar;
        public static string StaticVar { get; } = "I'm a static variable in Class1";

        public Class1()
        {
            privateVar = "I'm private";
        }

        public async Task Method1Async()
        {
            Console.WriteLine("This is Method1 from Class1");
            await PrivateMethodAsync();
        }

        private async Task PrivateMethodAsync()
        {
            Console.WriteLine("This is a private method in Class1");
            Console.WriteLine($"Private var: {privateVar}");
            Console.WriteLine($"Static var: {StaticVar}");
            await Task.Delay(100); // Simulate some async work
        }

        // Demonstrate generic method
        public T ProcessData<T>(T input) where T : IComparable<T>
        {
            Console.WriteLine($"Processing data of type {typeof(T).Name}");
            return input;
        }

        // Demonstrate yield return
        public IEnumerable<int> GenerateSequence(int count)
        {
            for (int i = 0; i < count; i++)
            {
                yield return i;
            }
        }

        // Demonstrate extension method
        public static class Extensions
        {
            public static string AddExclamation(this string str)
            {
                return $"{str}!";
            }
        }
    }
}